sdk/textanalytics/Azure.AI.TextAnalytics/src/TextAnalyticsClient.cs

```diff
@@ -246,7 +252,7 @@ public TextAnalyticsClient(Uri endpoint, AzureKeyCredential credential, TextAnal
         /// <exception cref="RequestFailedException">Service returned a non-success
         /// status code.</exception>
         public virtual Response<DetectLanguageResultCollection> DetectLanguageBatch(IEnumerable<string> documents, string countryHint = default, TextAnalyticsRequestOptions options = default, CancellationToken cancellationToken = default) =>
-            _serviceClient.DetectLanguageBatch(documents, countryHint, options, cancellationToken);
+            _serviceClient.DetectLanguageBatch(documents, countryHint, options?.CheckSupported(ServiceVersion), cancellationToken);
```

Comment:
Question about behavior.  It looks like this will throw a NotSupportedException if someone sets DisableServiceLogs to any value.  Is this behavior we think customers would want?  One of true/false must match the service default, so if a user set the value to the same value as the service default, they would change nothing about the operation but we would prevent this by throwing an exception.  This feels incorrect to me.  I would expect us to only prevent the request unless sending the property could somehow cause data corruption.  

As a general rule, I think we should only throw our own exception, and not the standard RequestFailedException with the service error, if we know the error message coming back from the service doesn't help the client user understand what they did wrong and how to fix it.  In a manually-written customization layer, I would use that to decide, and if we need to throw a post-facto exception to indicate a versioning issue caused the problem, I would add the service-side RequestFailedException as an inner exception.

Reply:
It's a good point but knowing how - or even if - the service will return a specific enough error, or whether the error is even an HTTP 400, will be difficult. Right now, our guidance for swaggers is to only document your 2xx response model and assume any error is `default`, without going into specifics. Conceptually, I love the idea but I think doing it in practice will be much harder.

Without hardcoding what the default is - and, of course, that could change (KV has done it) - we can't really know. So only doing this for error responses would make even more sense; but again, I worry about practical feasibility. And @tg-msft's [suggestion above](https://github.com/Azure/azure-sdk-for-net/pull/30753#discussion_r957717131) wouldn't change that. We'd just know if it is changed to any value. So, in practice, it makes me wonder if I should leave which, yes, is assuming the default remains the same; however, that does at least lend to the oft possibility that they are setting the value to its non-default value.

Reply:
Yeah, I just picked being redundant with the default of one case of sending a request that might still be useful if accepted by the service.  It's worth noting here, however, that if the service is following Azure Guidelines, they would reject any request that has unknown JSON values.

Overall, I think this is further argument for using post-facto validation rather than request validation.

Reply:
After testing, in TA's case they don't err for unsupported query parameters at all. After discussing in the v-team, we've decided that we will do client-side checks because a customer expecting any information based on setting this or other properties that are only supported in newer versions will be unpleasantly surprised.